pyvista/_overrides.py

```diff
@@ -0,0 +1,58 @@
+"""Functionality to override PyVista object creation with other subclasses."""
+from functools import wraps
+import inspect
+
+from pyvista.core import DataSet
+
+
+def set_up_overrides(root=DataSet):
+    """Set up type overriding for all subclasses of a PyVista type.
+
+    Setup entails recursively finding all subclasses of the root
+    type and defining an optional ``type_override`` class
+    attribute on each. __new__ is decorated for each type to
+    return the overriding type if provided.
+
+    Parameters
+    ----------
+    root : type, default: pyvista.DataSet
+        Parent type whose (recursive) subclasses will be set up.
+        Root itself is included.
+    """
+    for cls in OVERRIDABLES:
+        cls.type_override = None
+        old_new = cls.__new__
```

Comment:
I think it would be more readable to use "old" as "base" (naming from the base class) since it does override :)
```suggestion
        base_new = cls.__new__
```

Reply:
Or you may use `orig` to match helpers.py.
```suggestion
        orig_new = cls.__new__
```

Reply:
I thought it was funny to use this naming :smile: But thanks, you're right, it should be something clearer instead. I'll hold off a bit on choosing a new name, because `orig_new` in `helpers.py` is also something I just made up, so I'm not the best judge. We might want to rename both to something even better.